generate/convert.go

```diff
@@ -253,6 +261,18 @@ func (g *generator) convertType(
 	return goTyp, err
 }
 
+// getWeakReference decides if a field should be of pointer type and have the omitempty flag set.
+func (g *generator) getWeakReference(
+	options *genqlientDirective,
+	isStructField bool,
+	def *ast.Definition,
+) bool {
+	return options.Pointer == nil &&
```

Comment:
I think this will be a little confusing in that it will mean if you do `omitempty: false` it will be ignored, but if you do `pointer: false` it will not only work but also turn off `omitempty`.  (And in fact if you do `pointer: true` it will also turn off omitempty!)  If we can, I think it'll be better to check each setting separately, so that if you set `pointer` explicitly that overrides `pointer`, like you have it, but not `omitempty`.

Reply:
Yep, you're correct.  The logic is slightly flawed here as you describe.  I'll rework